vault/eventbus/bus.go

```diff
@@ -47,14 +58,27 @@ func (bus *EventBus) Start() {
 	}
 }
 
-// Send sends an event to the event bus and routes it to all relevant subscribers.
+// SendInternal sends an event to the event bus and routes it to all relevant subscribers.
 // This function does *not* wait for all subscribers to acknowledge before returning.
-func (bus *EventBus) Send(ctx context.Context, eventType logical.EventType, data *logical.EventData) error {
+// This function is meant to be used by trusted internal code, so it can specify details like the namespace
+// and plugin info. Events from plugins should be routed through WithPlugin(), which will populate
+// the namespace and plugin info automatically.
+func (bus *EventBus) SendInternal(ctx context.Context, namespace *namespace.Namespace, pluginInfo *logical.EventPluginInfo, eventType logical.EventType, data *logical.EventData) error {
 	if !bus.started.Load() {
 		return ErrNotStarted
 	}
-	bus.logger.Info("Sending event", "event", data)
-	_, err := bus.broker.Send(ctx, eventlogger.EventType(eventType), data)
+	var nspace string
+	if namespace != nil {
+		nspace = namespace.ID
+	}
+	eventReceived := &logical.EventReceived{
+		Event:      data,
+		Namespace:  nspace,
```

Comment:
It's good to use namespace ID internally for reliable equality checks, but I think we should also try to include namespace name for external presentation, as that's what most users interact with in the API. Perhaps we should have both fields in the event? Then we can continue to use ID everywhere internally, but also give it a human-friendly name that's easily understandable by external systems too.

Reply:
I think it makes sense to use the name when it is seen externally. I'll take a look to see what is the easiest way to keep the ID and the name.

Reply:
Based on the enterprise code, I think passing the path would be better than the ID.